File: frontend/src/services/jobService.js
import api from './api';

export const jobService = {
  // Public APIs
  searchJobs: async (params = {}) => {
    const response = await api.get('/jobs', { params });
    return response.data;
  },

  getJobById: async (id) => {
    const response = await api.get(`/jobs/${id}`);
    return response.data;
  },

  // Recruiter APIs
  createJob: async (jobData) => {
    const response = await api.post('/recruiter/jobs', jobData);
    return response.data;
  },

  getMyJobs: async (page = 0, size = 10) => {
    const response = await api.get('/recruiter/jobs', { params: { page, size } });
    return response.data;
  },

  updateJob: async (id, jobData) => {
    const response = await api.put(`/recruiter/jobs/${id}`, jobData);
    return response.data;
  },

  deleteJob: async (id) => {
    const response = await api.delete(`/recruiter/jobs/${id}`);
    return response.data;
  },

  getRecruiterProfile: async () => {
    const response = await api.get('/recruiter/profile');
    return response.data;
  },

  updateRecruiterProfile: async (profileData) => {
    const response = await api.put('/recruiter/profile', profileData);
    return response.data;
  },
};

export default jobService;

File: backend/src/main/java/com/talentbridge/service/JobService.java
package com.talentbridge.service;

import com.talentbridge.dto.JobCreateRequest;
import com.talentbridge.dto.JobResponse;
import com.talentbridge.dto.JobUpdateRequest;
import com.talentbridge.dto.RecruiterProfileDto;
import com.talentbridge.entity.ExperienceLevel;
import com.talentbridge.entity.JobType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface JobService {

    Page<JobResponse> searchJobs(String keyword, String location, JobType jobType,
                                 ExperienceLevel experienceLevel, Pageable pageable);

    JobResponse getJobById(Long id);

    JobResponse createJob(JobCreateRequest request, Long userId);

    JobResponse updateJob(Long id, JobUpdateRequest request, Long userId);

    void deleteJob(Long id, Long userId);

    Page<JobResponse> getRecruiterJobs(Long userId, Pageable pageable);

    RecruiterProfileDto getRecruiterProfile(Long userId);

    RecruiterProfileDto updateRecruiterProfile(Long userId, RecruiterProfileDto profileDto);
}
